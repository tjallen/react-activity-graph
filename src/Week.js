/* eslint-disable no-console */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Day from './Day';
import { v4 } from 'uuid';

export default class Week extends Component {
  static propTypes = {
    // data: PropTypes.array,
    color: PropTypes.string,
    week: PropTypes.any,
    maxValue: PropTypes.any,
  }
  render() {
    const styles = {
      display: 'inline-block',
      float: 'left',
    };
    const { week, maxValue, color } = this.props;
    // console.log(week);
    return (
      <div style={styles}>
        {week.map(item =>
          <Day
            key={v4()}
            value={item.value}
            date={item.date}
            maxValue={maxValue}
            color={color}
          >
            {item.value}
          </Day>
        )}
      </div>
    );
  }
}
