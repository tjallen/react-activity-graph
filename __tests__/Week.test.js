import React from 'react';
import Week from '../src/Week';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
configure({ adapter: new Adapter() });
import { format, subDays } from 'date-fns';

function fillWeek() {
  const week = [];
  const now = new Date();
  for (let i = 0; i < 7; i++) {
    const date = format(subDays(now, i), 'YYYY-MM-DD');
    week.unshift({
      date,
      value: i,
    });
  }
  return week;
}

const props = {
  week: fillWeek(),
  maxValue: 99,
  color: '#008000',
  bgColor: '#EBEDF0',
};

const component = renderer.create(
  <Week week={props.week} color={props.color} bgColor={props.bgcolor} />
);

// const wrapper = shallow(
//   <Week {...props} />
// );

test('Graph snapshot test', () => {
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
