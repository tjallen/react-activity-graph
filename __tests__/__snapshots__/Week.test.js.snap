// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph snapshot test 1`] = `
<div
  style={
    Object {
      "background": undefined,
      "display": "inline-block",
      "float": "left",
    }
  }
>
  <div
    className="sc-htpNat lcFUQH"
  >
    <div
      className="sc-bwzfXH kZCWKe"
    >
      <div
        className="sc-bdVaJa dOxDA"
      >
        6
         on 
      </div>
    </div>
  </div>
  <div
    className="sc-EHOje fqsyqh"
  >
    <div
      className="sc-ifAKCX vfOHM"
    >
      <div
        className="sc-bxivhb gQiCQK"
      >
        5
         on 
      </div>
    </div>
  </div>
  <div
    className="sc-htoDjs dpnqJM"
  >
    <div
      className="sc-gzVnrw eEDgMm"
    >
      <div
        className="sc-bZQynM jkROJx"
      >
        4
         on 
      </div>
    </div>
  </div>
  <div
    className="sc-gZMcBi gMNwBJ"
  >
    <div
      className="sc-iwsKbI dyMiCD"
    >
      <div
        className="sc-dnqmqq inxySn"
      >
        3
         on 
      </div>
    </div>
  </div>
  <div
    className="sc-jTzLTM gXCKra"
  >
    <div
      className="sc-VigVT gMiwNe"
    >
      <div
        className="sc-gqjmRU fSvBYA"
      >
        2
         on 
      </div>
    </div>
  </div>
  <div
    className="sc-cSHVUG fkgJpg"
  >
    <div
      className="sc-jzJRlG cfrftn"
    >
      <div
        className="sc-fjdhpX hsEdRP"
      >
        1
         on 
      </div>
    </div>
  </div>
  <div
    className="sc-kgoBCf gYjrLW"
  >
    <div
      className="sc-chPdSV hYJnEY"
    >
      <div
        className="sc-kAzzGY itfBWo"
      >
        0
         on 
      </div>
    </div>
  </div>
</div>
`;
